pkg/mcclient/mcclient.go

```diff
@@ -0,0 +1,158 @@
+package mcclient
+
+// kubestellar cluster-aware client impl
+
+import (
+	"context"
+	"errors"
+	"reflect"
+	"sync"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+	log "k8s.io/klog/v2"
+
+	"github.com/kcp-dev/edge-mc/pkg/apis/edge/v1alpha1"
+	ksclientset "github.com/kcp-dev/edge-mc/pkg/client/clientset/versioned"
+	ksinformers "github.com/kcp-dev/edge-mc/pkg/client/informers/externalversions"
+	mcclient "github.com/kcp-dev/edge-mc/pkg/mcclient/clientset"
```

Comment:
While this is well-defined in golang, in my opinion it is confusing to define an import package to have the same name as appears in the `package` statement.